### PR TITLE
Update editorconfig

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,32 +1,23 @@
...
-
-# Pom files
-[pom.xml]
-indent_size = 2
+insert_final_newline = true
+charset = utf-8-bom
 
 # Xml project files
-[*.{csproj,proj}]
+[*.{csproj,vbproj,vcxproj,vcxproj.filters,proj,projitems,shproj}]
```
Comment (valhristov) [inline .editorconfig]: Why is this necessary?

File: .editorconfig [hunk 2]
```diff
@@ -37,7 +28,28 @@ indent_size = 2
...
+
+# Dotnet code style settings:
+[*.{cs,vb}]
 # Sort using and Import directives with System.* appearing first
 dotnet_sort_system_directives_first = true
+# Avoid "this." and "Me." if not necessary
+dotnet_style_qualification_for_field = false:suggestion
+dotnet_style_qualification_for_property = false:suggestion
+dotnet_style_qualification_for_method = false:suggestion
+dotnet_style_qualification_for_event = false:suggestion
```
Comment (valhristov) [inline .editorconfig]: What about `var`? (troll)

File: .editorconfig [hunk 3]
```diff
@@ -1,32 +1,23 @@
...
-
-# Java code files
-[*.{java}]
-indent_size = 2
-
-# Pom files
-[pom.xml]
-indent_size = 2
+insert_final_newline = true
+charset = utf-8-bom
```
Comment (michalb-sonar) [inline .editorconfig]: Good point about BOM. We are very inconsistent with that.

File: .editorconfig (hunk 1 above)
Comment (Evangelink) [reply .editorconfig]: Just copy-pasting what is done on roslyn ;)

File: .editorconfig (hunk 2 above)
Comment (Evangelink) [reply .editorconfig]: Sadly it doesn't work with VS 2015, I did planned to add this discreetly...
